### PR TITLE
Add monkey script

### PR DESCRIPTION
**Pull request by demerzel3**
Ref. #7

### CODE REVIEW

**Comment by demerzel3** [comment]
So.. I added the cancel feature, the auto-remove on merge button click, the polling interval for auto-update of the position in the queue and a couple of fixes here and there.
I guess we can go on to some field testing! What so you think @maxcanna?

**Comment by maxcanna** [comment]
I think that you did a great job @demerzel3! We can definitely start with the internal usage to have a rough test of this and #10. We'll have a chance to test their interaction with the backend in a real scenario.

**Comment by demerzel3** [comment]
Possible improvements:
- ask for confirmation before cancelling
- warn if the user booking their place to merge is not an assignee of the PR (may be booking the wrong one and not noticed)

**Comment by giorrrgio** [comment]
The button is not displayed in Linux/Firefox 49.02 with greasemonkey 3.9

File: monkey/branch-bookkeeper.user.js [hunk 1]
```diff
@@ -5,6 +5,7 @@
 // @description  Enable bookkeeping
 // @author       AdEspresso
 // @include      /^https://github.com/adespresso/adespresso/pull/\d+/
+// @include      /^https://branch-bookkeeper.herokuapp.com/
```
Comment (demerzel3) [inline monkey/branch-bookkeeper.user.js]: We should support the homepage via `http` as well.

File: monkey/branch-bookkeeper.user.js [hunk 2]
```diff
@@ -286,5 +321,7 @@ function prerequisitesChecksDecorator(canProceed, innerFunction) {
 }
 
 const decoratedMain = prerequisitesChecksDecorator(canProceed, main);
+const decoratedHome = prerequisitesChecksDecorator(HomePageModule.isHomePage, HomePageModule.showRightMessage);
 
 (() => waitForKeyElements('.merge-message', decoratedMain))();
+(() => waitForKeyElements('#script-ko', decoratedHome))();
```
Comment (demerzel3) [inline monkey/branch-bookkeeper.user.js]: Once we detect being on github or on the homepage we should stop looking for elements of the other page or the `waitForKeyElements` will try forever.

Now that I think about it, a simple check on `window.location` should be enough to distinguish the two cases.

File: monkey/branch-bookkeeper.user.js (hunk 1 above)
Comment (maxcanna) [inline monkey/branch-bookkeeper.user.js]: We're gonna redirect `http` connections to `https` isn't it?

File: monkey/branch-bookkeeper.user.js (hunk 2 above)
Comment (mattiaocchiuto) [inline monkey/branch-bookkeeper.user.js]: Done [here](https://github.com/adespresso/branch-bookkeeper/pull/9/commits/53c29047fa86f99053aada9e56a198576d891fbd)

**Comment by mattiaocchiuto** [comment]
We should enable/disable the merge button using a custom class instead of adding/removing the disabled attribute

**Comment by maxcanna** [comment]
We've to dynamically get constants (`OWNER`, `REPOSITORY` and `BRANCH`) from GH interface.
`API_BASE_URL` and `HOME_PAGE_HOST` will be valued server side while `GITHUB_PAGE_HOST` can be left as it is.

**Comment by maxcanna** [comment]
Done in #14

**Comment by maxcanna** [comment]
> The button is not displayed in Linux/Firefox 49.02 with greasemonkey 3.9

It looks like that using greasemonkey we've some problem with loading of Lodash in combination with Backbone.js. The latter won't see the first one if both are present. Such weird.

**Comment by demerzel3** [comment]
Since the proposed `requirejs` solution is not going to work because of cross domain issues, we should probably try the hacky way: just throw a `<script>` tag in the page for each script dependency. If this works (and apart from being ugly) this could also simplify the manifest file we use in development.

**Comment by demerzel3** [comment]
TL;DR Firefox issue fixed

Details: the ugly fix didn't work because github does not allow scripts from other domains (which is probably wise..); the script in Firefox's tampermonkey does not work (still no idea why); I managed to make it work under greasemonkey by tampering with Backbone version: I knew from previous experience that Backbone changed the way it looks for underscore from version `1.1.x` to `1.2.x`, I tried it and it is working 🍾 , we are now relaying on version `1.1.2` instead of `1.3.3`.
